.pre-commit-config.yaml

```diff
@@ -0,0 +1,12 @@
+repos:
```

Comment:
Need to add to development docs how to set it up. Also, for those of us who aren't fans of pre-commit hooks, I recommend that we make sure that the top IDEs can be setup to do all of these checks automatically and provide instructions on how to configure it so (not in this PR though)